### PR TITLE
Reenable warning C4463, C4311, C4499 and C4091

### PR DESCRIPTION
**Pull request by curtisman**
Fix and reenable these warnings:
C4091 - ignore on left of '<type>' when no variable is declared
C4311 - 'type cast': pointer truncation from '' to ''
C4463  - overflow; assigning 1 to bit-field that can only hold values from -1 to 0
C4499 - explicit specialization cannot have a storage class (ignored)

### CODE REVIEW

File: lib/Parser/Scan.h [hunk 1]
```diff
@@ -667,7 +667,7 @@ class Scanner : public IScanner, public EncodingPolicy
     BOOL m_doubleQuoteOnLastTkStrCon :1;
     bool m_OctOrLeadingZeroOnLastTKNumber :1;
     BOOL m_fSyntaxColor : 1;            // whether we're just syntax coloring
```
Comment (agarwal-sandeep) [inline lib/Parser/Scan.h]: Should all BOOL be changed to bool?

**Comment by curtisman** [comment]
@digitalinfinity please review.

File: lib/Parser/Scan.h (hunk 1 above)
Comment (curtisman) [inline lib/Parser/Scan.h]: Yeah... that get rids of the warning.
I guess the rest of the fields should be change to bool too..... but those didn't cause warnings, so I didn't touch it.

**Comment by dilijev** [comment]
LGTM

File: lib/Parser/Scan.h (hunk 1 above)
Comment (ianwjhalliday) [inline lib/Parser/Scan.h]: This pattern of mixing types on bit fields might be bloating our struct sizes: http://stackoverflow.com/questions/308364/c-bitfield-packing-with-bools

**Pull request merged**